mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1604,14 +1619,20 @@ pub async fn gen_and_sign_taker_payment_spend_preimage<T: UtxoCommonOps>(
     let preimage = gen_taker_payment_spend_preimage(coin, args, NTimeSetting::UseNow).await?;
 
     let redeem_script =
-        swap_proto_v2_scripts::taker_payment_script(time_lock, args.secret_hash, args.taker_pub, args.maker_pub);
+        swap_proto_v2_scripts::taker_payment_script(time_lock, args.maker_secret_hash, args.taker_pub, args.maker_pub);
+
+    let sig_hash_type = match args.dex_fee {
+        DexFee::Standard(_) => SIGHASH_SINGLE,
+        DexFee::WithBurn { .. } => SIGHASH_ALL,
```

Comment:
I assume that it will take some time to figure out the way on how dex fee burn will work for other coins, so I chose the simplest implementation way for the moment :slightly_smiling_face:

Reply:
Sure, I think having the 2 options is good and will help us implement dex fee burn gradually for other coins. I just wanted to bring this to your attention in case there are other considerations that I missed.